src/Node/Stream.purs

```diff
@@ -70,104 +84,156 @@ type Writable r = Stream (write :: Write | r)
 -- | A duplex (readable _and_ writable stream)
 type Duplex = Stream (read :: Read, write :: Write)
 
-foreign import undefined :: forall a. a
+toEventEmitter :: forall rw. Stream rw -> EventEmitter
+toEventEmitter = unsafeCoerce
 
+-- | Internal type. This should not be used by end-users.
 foreign import data Chunk :: Type
 
 foreign import readChunkImpl
-  :: (forall l r. l -> Either l r)
-  -> (forall l r. r -> Either l r)
-  -> Chunk
-  -> Either String Buffer
-
-readChunk :: Chunk -> Either String Buffer
-readChunk = readChunkImpl Left Right
+  :: forall r
+   . EffectFn3
+       (EffectFn1 Buffer r)
+       (EffectFn1 String r)
+       Chunk
+       r
 
 -- | Listen for `data` events, returning data in a Buffer. Note that this will fail
 -- | if `setEncoding` has been called on the stream.
-onData
-  :: forall w
-   . Readable w
-  -> (Buffer -> Effect Unit)
-  -> Effect Unit
-onData r cb =
-  onDataEither r (cb <=< fromEither)
-  where
-  fromEither x =
-    case x of
-      Left _ ->
-        throw "Stream encoding should not be set"
-      Right buf ->
-        pure buf
-
-read
-  :: forall w
-   . Readable w
-  -> Maybe Int
-  -> Effect (Maybe Buffer)
-read r size = do
-  v <- readEither r size
-  case v of
-    Nothing -> pure Nothing
-    Just (Left _) -> throw "Stream encoding should not be set"
-    Just (Right b) -> pure (Just b)
-
+-- |
+-- | This is likely the handler you want to use for converting a `Stream` into a `String`:
+-- | ```
+-- | let useStringCb = ...
+-- | ref <- Ref.new
+-- | stream # on dataH \buf ->
+-- |   Ref.modify_ (\ref' -> Array.snoc ref' buf) ref
+-- | stream # on endH do
+-- |   bufs <- Ref.read ref
+-- |   useStringCb $ Buffer.toString UTF8 $ Buffer.concat bufs
+-- | ```
+dataH :: forall w. EventHandle (Readable w) (Buffer -> Effect Unit) (EffectFn1 Chunk Unit)
```

Comment:
For a subjective reason, using the `H` suffix means the entire function and its callback can be put on one line. For example
```
emitter # on eventNameHSpecialVariant \arg1 arg arg3 arg4 ->
  ...
```
Instead of
```
emitter # on eventNameHandlerSpecialVariant
  \arg1 arg arg3 arg4 ->
     ...
```

Reply:
> For a subjective reason, using the H suffix means the entire function and its callback can be put on one line.

I don't see why the second example here has to be dropped to a new line.

> No, there is not. But, I'm not willing to change that.

OK. Well, I don't have other feedback, so I suppose you're good to go — you don't strictly need an approval (there was no approval on the 3.0.0 release of `node-event-emitter` either, where these names were introduced).

Reply:
> you don't strictly need an approval (there was no approval on the 3.0.0 release of node-event-emitter either, where these names were introduced).

I assumed I didn't need approval because I wrote the `node-event-emitter` library. :shrug:.

As for these libraries, more people use them and a second pair of eyes will help ensure no new problems are introduced by my changes. But, if that's something people don't care about, then I suppose I could just update all of these libraries without any approval on the resulting PRs.